cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/BulkWriteValidator.java

```diff
@@ -110,19 +112,24 @@ public static void updateFailureHandler(CommitResult commitResult,
         });
     }
 
-    private static void logFailedRanges(Logger logger, String phase,
-                                        List<ReplicaAwareFailureHandler<RingInstance>.ConsistencyFailurePerRange> failedRanges)
+    // aggregate the stream errors in order to provide a better insight on failure
+    private static String aggregateErrors(List<ReplicaAwareFailureHandler<RingInstance>.ConsistencyFailurePerRange> failedRanges)
     {
+        StringBuilder sb = new StringBuilder();
         for (ReplicaAwareFailureHandler<RingInstance>.ConsistencyFailurePerRange failedRange : failedRanges)
         {
-            failedRange.failuresPerInstance.forEachInstance((instance, errors) -> {
-                logger.error("Failed in phase {} for {} on {}. Failure: {}",
```

Comment:
Not thrilled with the fact that we lost the logging completely - I'd build the error message/cap its length, but continue logging everything, rather than just building the error and returning it.

Reply:
The logging is moved to `validateClOrFail`. However, I should probably not truncate the message here. Making an update.